### PR TITLE
Add config instructions for HTTPS on Karaf.

### PR DESCRIPTION
**Pull request by geomacy**
Plus a few more details on key generation.

### CODE REVIEW

**Comment by drigodwin** [comment]
Is `brooklyn_properties.md` the best place for this considering it's not in `brooklyn.properties`?

**Comment by geomacy** [comment]
Well it's configuration, and is together with the other HTTPS stuff; I guess I could move the Karaf bits to the "osgi" page, and cross link each to the other?  We could leave out the cross links but then it's maybe hard for someone to know where to find the info they need? What do you think?

**Comment by drigodwin** [comment]
I'd be tempted to just make a new HTTPS Configuration page.

**Comment by geomacy** [comment]
@drigodwin I've updated with a new page as you suggested, in fact I've done a bit of a reorganisation, can you have a look and see what you think, I think this is a nicer structure.

File: guide/ops/brooklyn_properties.md [hunk 1]
```diff
@@ -211,23 +211,10 @@ or
 {% include java_link.html class_name="EntitlementManager" package_path="org/apache/brooklyn/api/mgmt/entitlement" project_subpath="api" %}.
 
 
-## HTTPS Configuration
 
-To enable https, you will need a server certificate in a java keystore. To create a self-signed certificate, you can use the
-following command:
+## Configuring HTTPS in Brooklyn
```
Comment (drigodwin) [inline guide/ops/brooklyn_properties.md]: I think you need to change `section: HTTPS Configuration` at the top too if you're going to change the titles.

File: guide/ops/https.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+---
+title: HTTPS Configuration
+layout: website-normal
+---
+
+## HTTPS Configuration
+
+### Getting the Certificate
+To enable HTTPS web access, you will need a server certificate in a java keystore. To create a self-signed certificate,
```
Comment (drigodwin) [inline guide/ops/https.md]: Perhaps link out [java keystore](https://en.wikipedia.org/wiki/Keystore) and say something like - To create a self-signed certificate (for testing and non-production use)

File: guide/ops/https.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+and add it to a keystore, `keystore.jks`, you can use the following command:
+
+{% highlight bash %}
+% keytool -genkey -keyalg RSA -alias brooklyn \
+          -keystore <path-to-keystore-directory>/keystore.jks -storepass "mypassword" \
+          -validity 365 -keysize 2048 -keypass "password"
+{% endhighlight %}
+
+Of course, the passwords above should be changed.  Omit those arguments above for the tool to prompt you for the values.
```
Comment (drigodwin) [inline guide/ops/https.md]: Perhaps change "Of course, the passwords above should be changed" to "The passwords above should be changed to something [more secure](https://en.wikipedia.org/wiki/Password_strength)"

File: guide/ops/https.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+You will then be prompted to enter your name and organization details. This will use (or create, if it does not exist)
+a keystore with the password `mypassword` - you should use your own secure password, which will be the same password
+used in your brooklyn.properties (below). You will also need to replace `<path-to-keystore-directory>` with the full 
+path of the folder where you wish to store your keystore. The keystore will contain the newly generated key, 
+with alias `brooklyn` and password `password`.
+
+The certificate generated will be a self-signed certificate, which will cause a warning to be displayed by a browser
```
Comment (drigodwin) [inline guide/ops/https.md]: Move this paragraph to the top of this section?

File: guide/ops/osgi-configuration.md [hunk 5]
```diff
@@ -1,79 +1,62 @@
 ---
-title: OSGi Distribution
+title: OSGi Configuration
 layout: website-normal
 ---
 
-# Running Apache Brooklyn inside Karaf container
+Configuration of Brooklyn when running under Karaf is largely done through standard Karaf mechanisms.
```
Comment (drigodwin) [inline guide/ops/osgi-configuration.md]: You should refer to at least the first instance of Brooklyn within a document as Apache Brooklyn

File: guide/ops/osgi-configuration.md [hunk 6]
```diff
@@ -1,79 +1,62 @@
...
-title: OSGi Distribution
+title: OSGi Configuration
 layout: website-normal
 ---
 
-# Running Apache Brooklyn inside Karaf container
+Configuration of Brooklyn when running under Karaf is largely done through standard Karaf mechanisms. 
+The Karaf "Configuration Admin" subsystem is used to manage configuration values loaded at first boot from the
+`.cfg` files in the `etc` directory of the distribution. In the Karaf command line these can then be viewed
+and manipulated by the `config:` commands, see the Karaf documentation for full details.
```
Comment (drigodwin) [inline guide/ops/osgi-configuration.md]: Link out to the Karaf documentation

File: guide/ops/osgi-configuration.md [hunk 7]
```diff
@@ -1,79 +1,62 @@
...
-JAVA_DEBUG_OPTS='-Xdebug -Xnoagent -Djava.compiler=NONE -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=5005' bin/karaf debug
-{% endhighlight %}
-
-The Karaf container will keep state like installed bundles and configuration between restarts.
-To reset any changes add **clean** to the cli arguments.
-
-## Configuring
-
-To configure the Brooklyn runtime create an **etc/brooklyn.cfg** file, following the standard **brooklyn.properties**
+To configure the Brooklyn runtime create an `etc/brooklyn.cfg` file, following the standard `brooklyn.properties`
```
Comment (drigodwin) [inline guide/ops/osgi-configuration.md]: I think the ** links out to the file? This should link out to the brooklyn.properties page I think

File: guide/ops/osgi.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+title: OSGi Distribution
+layout: website-normal
+children:
+- osgi-configuration.md
+---
+
+# Running Apache Brooklyn inside Karaf container
+
+The Apache Brooklyn Karaf based distribution lives in brooklyn-server/karaf/apache-brooklyn folder.
+It's still in a testing stage so some features might not work as expected. Do reach out if you
```
Comment (drigodwin) [inline guide/ops/osgi.md]: I don't think I would use a colloquialism like 'reach out' here

File: guide/ops/osgi.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+where you can interact with the running instance. Note that Brooklyn has already started at this point
+and is reachable at the usual web console url.
+
+To start in debug mode use
+
+{% highlight bash %}
+bin/karaf debug
+{% endhighlight %}
+
+and connect to port 5005 using your debugger.
```
Comment (drigodwin) [inline guide/ops/osgi.md]: Whats the debugger? - perhaps link out or describe this further

File: guide/ops/osgi.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+To pause startup until the debugger is connected you can use
+
+{% highlight bash %}
+JAVA_DEBUG_OPTS='-Xdebug -Xnoagent -Djava.compiler=NONE -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=5005' bin/karaf debug
+{% endhighlight %}
+
+The Karaf container will keep state like installed bundles and configuration between restarts.
```
Comment (drigodwin) [inline guide/ops/osgi.md]: keep state such as installed

File: guide/ops/https.md (hunk 2 above)
Comment (geomacy) [inline guide/ops/https.md]: done, although I don't think we need to link out to docs for the tool.

File: guide/ops/https.md (hunk 3 above)
Comment (geomacy) [inline guide/ops/https.md]: done

File: guide/ops/brooklyn_properties.md (hunk 1 above)
Comment (geomacy) [inline guide/ops/brooklyn_properties.md]: sorted

File: guide/ops/https.md (hunk 4 above)
Comment (geomacy) [inline guide/ops/https.md]: done

File: guide/ops/osgi-configuration.md (hunk 5 above)
Comment (geomacy) [inline guide/ops/osgi-configuration.md]: done

File: guide/ops/osgi-configuration.md (hunk 6 above)
Comment (geomacy) [inline guide/ops/osgi-configuration.md]: done

File: guide/ops/osgi-configuration.md (hunk 7 above)
Comment (geomacy) [inline guide/ops/osgi-configuration.md]: the ** just makes it bold, think the backticks is more consistent

File: guide/ops/osgi.md (hunk 8 above)
Comment (geomacy) [inline guide/ops/osgi.md]: fixed and added link to dev@brooklyn.apache.org

File: guide/ops/osgi.md (hunk 9 above)
Comment (geomacy) [inline guide/ops/osgi.md]: tweaked for a bit of extra clarity

File: guide/ops/osgi.md (hunk 10 above)
Comment (geomacy) [inline guide/ops/osgi.md]: done

**Comment by neykov** [comment]
Thanks @geomacy, @drigodwin, merging.

**Comment by geomacy** [comment]
Thanks @neykov!

**Pull request merged**